comm/src/main/scala/coop/rchain/comm/unicast.scala

```diff
@@ -53,22 +55,21 @@ class UnicastComm(local: PeerNode) extends Comm[SocketAddress] {
       val payload = new Array[Byte](2 + sz)
       payload(0) = (sz >> 8).toByte
       payload(1) = sz.toByte
-      Array.copy(data.toArray, 0, payload.toArray, 2, sz)
-      sendBytes.increment(sz + 2)
+      Array.copy(data.toArray, 0, payload, 2, sz)
+      sendBytes.increment(sz.toLong + 2)
       Right(payload)
     }
   }
 
   private def decode(data: Array[Byte]): Either[CommError, Seq[Byte]] =
-    if (data.size < 2) {
-      Left(DatagramSizeError(data.size))
+    if (data.length < 2) {
```

Comment:
Out of curiosity, what's the difference?

Reply:
See http://www.scala-lang.org/files/archive/spec/2.11/12-the-scala-standard-library.html#class-array

`length` uses underlying JVM `array.length`. Whereas `size` performs a `Predef` implicit conversion to `ArrayOps` and then delegates to `length`